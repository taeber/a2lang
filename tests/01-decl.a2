var foo: int
use CH: byte @ $24
var CH2: byte @ -17
use CH3:        byte @ %10
var CH4: byte @ %10

let bool = :byte

var PTR1: addr @ $06
use PTR2: addr @ $FE

var []
var [one: byte ]

use [
    first : byte,
    sec   : char,
    third: addr,
]

var [
    x: char
]

use Copy: sub
    <- [dst: text @ PTR1, src: text @ PTR2]
    -> [success: bool]

use SafeCopy: sub <- [
    dst   :text @ PTR1,
    src   :text @ PTR2,
    limit :byte,
] -> [
    success: bool
]

let Point3D = [x: int, y: int, z: int]
let cNumPoints = 2
var points: Point3D^42
var points2: Point3D^cNumPoints

