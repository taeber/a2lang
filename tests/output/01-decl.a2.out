CH	EQU $24
CH2	EQU $FFEF
CH3	EQU $02
CH4	EQU $02
PTR1	EQU $06
PTR2	EQU $FE
Copy.dst	EQU PTR1
Copy.src	EQU PTR2
SafeCopy.dst	EQU PTR1
SafeCopy.src	EQU PTR2
cNumPoints	EQU $02
foo	HEX 00
one	HEX 00
x	HEX 00
Copy.success	HEX 00
SafeCopy.limit	HEX 00
SafeCopy.success	HEX 00
points	HEX 00000000000000000000000000000000
	HEX 00000000000000000000000000000000
	HEX 00000000000000000000000000000000
	HEX 00000000000000000000000000000000
	HEX 00000000000000000000000000000000
	HEX 00000000000000000000000000000000
	HEX 00000000000000000000000000000000
	HEX 0000000000000000000000000000
points2	HEX 000000000000
